sources/packages/api/src/services/form/form.service.ts

```diff
@@ -119,14 +130,48 @@ export class FormService {
    * @returns header to be added to HTTP request.
    */
   private async createAuthHeader() {
-    const token = await this.getAuthToken();
+    const token = await this.tryGetCachedToken();
     return {
       headers: {
-        [FORMIO_TOKEN_NAME]: token,
+        [FORMIO_TOKEN_NAME]: token.token,
       },
     };
   }
 
+  /**
+   * Cached token for Form.IO service, managed
+   * by the method tryGetCachedToken.
+   */
+  private cachedToken: {
+    token: any;
+    decoded: any;
+  };
+
+  /**
+   * Tries get the token from the cache if it is not expired
+   * or about to be expired, otherwise get a new one.
+   * @returns Cached token or a new one.
+   */
+  private async tryGetCachedToken(): Promise<{ token: any; decoded: any }> {
+    const needRenewToken =
+      !this.cachedToken || // If the token is not present, retrieve it.
+      needRenewJwtToken(
+        +this.cachedToken.decoded["exp"],
```

Comment:
there is a `+` before this.cachedToken.decoded["exp"].

Reply:
This is to convert the string to a number in a "shortcut" way.
https://www.codegrepper.com/code-examples/typescript/how+to+convert+string+to+number+in+typescript

![image](https://user-images.githubusercontent.com/61259237/122483328-2d270a80-cf87-11eb-97b7-49029853316c.png)